conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ContentTypeFilterHandler.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.logsafe.SafeArg;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import java.util.Set;
+
+public final class ContentTypeFilterHandler implements HttpHandler {
+
+    private final Set<String> acceptedContentTypes;
+    private final HttpHandler delegate;
+
+    public static ContentTypeFilterHandler of(Set<String> acceptedContentTypes, HttpHandler delegate) {
+        return new ContentTypeFilterHandler(acceptedContentTypes, delegate);
+    }
+
+    private ContentTypeFilterHandler(Set<String> acceptedContentTypes, HttpHandler delegate) {
+        this.acceptedContentTypes = acceptedContentTypes;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
+        String contentType = ContentTypes.getContentType(exchange);
+        if (acceptedContentTypes.contains(contentType)) {
```

Comment:
This is a bit tricky because a json consumer may consume `application/json` or `application/json; charset=UTF-8` or `application/json;charset=UTF-8`, even `application/json; foo=bar; baz=bang`

Reply:
Ya, I could see a case for a few different utility classes, something like this could be fairly generic (with no support for consuming an optional request body, but I think that's fine, it's rare and can be handled by implementing a DeserializerFactory+Deserializer)


```java
public interface SimpleDeserializer<U> extends DeserializerFactory<U> {

    boolean accepts(String contentType);

    U deserialize(InputStream content) throws IOException;

    @Override
    @SuppressWarnings("unchecked")
    default <T extends U> Deserializer<T> deserializer(TypeMarker<T> type, UndertowRuntime runtime, Endpoint endpoint) {
        return (Deserializer<T>) new Deserializer<U>() {
            @Override
            public U deserialize(HttpServerExchange exchange) throws IOException {
                String maybeContentType = exchange.getRequestHeaders().getFirst(Headers.CONTENT_TYPE);
                if (maybeContentType == null || !accepts(maybeContentType)) {
                    throw new SafeIoException("Unsupported content-type", SafeArg.of("contentType", maybeContentType));
                }
                return SimpleDeserializer.this.deserialize(exchange.getInputStream());
            }
        };
    }
}
```

we could provide a subclass which returns a content-type, where we do a best-effort match (`accepts` does the `startsWith` check)
The Serializer is a bit more complicated since there's some amount of negotiation based on the `Accept` value, but that can largely be ignored in most cases because it doesn't matter if we fail based on a missing Accept header, or write data that causes the client to fail -- in either case the client must also validate the Content-Type of the response.

Reply:
I like the idea of just providing a utility class that consumers can use. Updated the PR based on this suggestion.

Example for how an implementation might look like:
```java
enum WildcardDeserialzerFactory implements ContentTypeAcceptingDeserializerFactory<InputStream> {
    INSTANCE;

    @Override
    public String supportedContentType() {
        return MediaType.WILDCARD;
    }

    @Override
    public InputStream deserialize(HttpServerExchange exchange) {
        return exchange.getInputStream();
    }
}
```

Reply:
We have some internal use cases can consume multiple content types:
```java
@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
```

Updated the utils class to provide a list of accepted content types instead.